### PR TITLE
[RPA-1523] Bots and Bots collection migration

### PR DESCRIPTION
**Pull request by socjalis**
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

---

I agree to the terms of the RunBotics Contributor License Agreement.

### CODE REVIEW

File: runbotics/runbotics-orchestrator-ui/src/src-app/store/slices/Bot/Bot.thunks.ts [hunk 1]
```diff
@@ -3,33 +3,20 @@ import { IBot, NotificationBot, NotificationBotType } from 'runbotics-common';
...
-        .then((response) => response.data),
+    BOT_PATH,
 );
 
-export const getPage = createAsyncThunk<Page<IBot>, PageRequestParams<IBot>>(
+export const getPage = ApiTenantResource.get<Page<IBot>>(
     'bots/page',
-    (params) => Axios.get<Page<IBot>>(botPageURL(params))
-        .then((response) => response.data),
+    `${BOT_PATH}/Page`,
```
Comment (marcin-depcik) [inline runbotics/runbotics-orchestrator-ui/src/src-app/store/slices/Bot/Bot.thunks.ts]: change to `GetPage`

File: runbotics/runbotics-orchestrator-ui/src/src-app/store/slices/BotCollections/BotCollections.thunks.ts [hunk 2]
```diff
@@ -1,57 +1,32 @@
...
-    .url('/api/bot-collection-page/current-user')
-    .params(params)
-    .build();
-
-export const getByPage = createAsyncThunk<Page<IBotCollection>, PageRequestParams<Partial<IBotCollectionWithFilters>>>(
+export const getByPage = ApiTenantResource.get<Page<IBotCollection>>(
     'botCollection/getPageForUser',
-    (params) => axios.get<Page<IBotCollection>>(processPageURL(params))
-        .then((response) => response.data),
+    `${BOT_COLLECTIONS_PATH}/current-user/Page`,
```
Comment (marcin-depcik) [inline runbotics/runbotics-orchestrator-ui/src/src-app/store/slices/BotCollections/BotCollections.thunks.ts]: same

File: runbotics/runbotics-orchestrator-ui/src/src-app/utils/ApiTenantResource.ts [hunk 3]
```diff
@@ -58,7 +58,7 @@ class ApiTenantResource {
                     .catch(error => thunkApi.rejectWithValue(error.response.data));
             }
         );
-
+
```
Comment (marcin-depcik) [inline runbotics/runbotics-orchestrator-ui/src/src-app/utils/ApiTenantResource.ts]: remove

File: runbotics/runbotics-scheduler/src/scheduler-database/bot-collection/bot-collection.service.ts [hunk 4]
```diff
@@ -0,0 +1,273 @@
...
+        options.where = {
+            ...specs.where,
+            tenantId: user.tenantId,
+        };
+
+        options.order = specs.order;
+        options.relations = RELATIONS;
+
+        this.botCollectionRepository.find(options);
+    }
```
Comment (marcin-depcik) [inline runbotics/runbotics-scheduler/src/scheduler-database/bot-collection/bot-collection.service.ts]: what is the point if this function returns nothing?

File: runbotics/runbotics-scheduler/src/scheduler-database/bot-collection/bot-collection.service.ts [hunk 5]
```diff
@@ -0,0 +1,273 @@
...
+                        {
+                            name: specs.where.name ?
+                                And(Equal(DefaultCollections.GUEST), specs.where.name as FindOperator<string>)
+                                : DefaultCollections.GUEST,
+                        },
+                        {
+                            name: specs.where.name ?
+                                And(Equal(DefaultCollections.GUEST), specs.where.name as FindOperator<string>)
+                                : DefaultCollections.GUEST,
+                        },
```
Comment (marcin-depcik) [inline runbotics/runbotics-scheduler/src/scheduler-database/bot-collection/bot-collection.service.ts]: is this intentional?

File: runbotics/runbotics-scheduler/src/scheduler-database/bot-collection/bot-collection.service.ts [hunk 6]
```diff
@@ -0,0 +1,273 @@
...
+                {
+                    name: specs.where.name ?
+                        And(Equal(DefaultCollections.GUEST), specs.where.name as FindOperator<string>)
+                        : DefaultCollections.GUEST,
+                },
+                {
+                    name: specs.where.name ?
+                        And(Equal(DefaultCollections.GUEST), specs.where.name as FindOperator<string>)
+                        : DefaultCollections.GUEST,
+                },
```
Comment (marcin-depcik) [inline runbotics/runbotics-scheduler/src/scheduler-database/bot-collection/bot-collection.service.ts]: same as above

File: runbotics/runbotics-scheduler/src/scheduler-database/bot-collection/bot-collection.service.ts [hunk 7]
```diff
@@ -0,0 +1,273 @@
...
```
Comment (marcin-depcik) [inline runbotics/runbotics-scheduler/src/scheduler-database/bot-collection/bot-collection.service.ts]: same as above

File: runbotics/runbotics-scheduler/src/scheduler-database/bot/bot-crud.service.ts [hunk 8]
```diff
@@ -0,0 +1,103 @@
...
+
+    async findOne(user: UserEntity, id: number) {
+        const collectionIds = await this.botCollectionService.findIds(
+            user,
+            { where: {}, order: { created: 'asc' } },
+            { take: 1000, skip: 0 },
+        );
+        
+        const bot = await this.botRepository
+            .findOneBy({ tenantId: user.tenantId, id })
```
Comment (marcin-depcik) [inline runbotics/runbotics-scheduler/src/scheduler-database/bot/bot-crud.service.ts]: `findOneOrFail` should be used here

Comment (marcin-depcik) [inline runbotics/runbotics-scheduler/src/scheduler-database/bot/bot-crud.service.ts]: `findOneByOrFail` should be used here

File: runbotics/runbotics-scheduler/src/scheduler-database/process/process.controller.ts [hunk 9]
```diff
@@ -178,7 +179,7 @@ export class ProcessController {
     @FeatureKeys(FeatureKey.PROCESS_LIST_READ)
     getPage(
         @Specifiable(ProcessCriteria) specs: Specs<ProcessEntity>,
-        @Pageable() paging: Paging,
+        @Pageable() paging: Paging,
```
Comment (marcin-depcik) [inline runbotics/runbotics-scheduler/src/scheduler-database/process/process.controller.ts]: remove

File: runbotics/runbotics-scheduler/src/scheduler-database/process/process.entity.ts [hunk 10]
```diff
@@ -52,10 +52,10 @@ export class ProcessEntity {
     @Column('text')
     definition: string;
 
-    @CreateDateColumn({ transformer: dateTransformer, type: 'timestamp without time zone' })
+    @CreateDateColumn({ type: 'timestamp without time zone' })
     created: string;
 
-    @UpdateDateColumn({ transformer: dateTransformer, type: 'timestamp without time zone' })
+    @UpdateDateColumn({ type: 'timestamp without time zone' })
     updated: string;
```
Comment (marcin-depcik) [inline runbotics/runbotics-scheduler/src/scheduler-database/process/process.entity.ts:60]: to verify

File: runbotics/runbotics-scheduler/src/utils/page/page.ts [hunk 11]
```diff
@@ -15,19 +16,31 @@ export interface Page<T> {
...
 
     return {
         content: entities,
         empty: entities.length === 0,
-        first: options.skip === 0,
+        first: !skip || skip === 0,
         totalElements: count,
         totalPages,
-        last: totalPages === 0 || page === totalPages - 1,
+        last: page === totalPages - 1,
```
Comment (marcin-depcik) [inline runbotics/runbotics-scheduler/src/utils/page/page.ts]: to verify

File: runbotics/runbotics-scheduler/src/utils/specification/filter/string-filter/string.filter.ts [hunk 12]
```diff
@@ -28,9 +29,13 @@ export class StringFilter extends Filter {
         this.predicates.push(Not(ILike(`%${value}%`)));
     }
 
+    in(values: string) {
+        this.predicates.push(In(values.split(',')));
+    }
```
Comment (marcin-depcik) [inline runbotics/runbotics-scheduler/src/utils/specification/filter/string-filter/string.filter.ts:34]: I would move it to separate filter class `ListFilter`

**Review by marcin-depcik — APPROVED** [review]

**Review by AdriannaSuchomska — APPROVED** [review]

**Pull request merged**